docs/enhancements/install-strategy.md

```diff
@@ -0,0 +1,196 @@
+
+AgentInstallStrategy becomes a day 1 resource and will offer more flexibility for the agent controllers to define settings they need in isolation, and track their relevant status.
+
+
+### User Stories
+
+#### Story 1
+
+#### Story 2
+
+## Design Details
+
+
+### ClusterDeployment Changes
+
+ClusterDeployment should now contain only information needed to connect to the cluster and settings for day 2 management. (admin kubeconfig, admin password, ingress, control plane certs, etc) Platform section will remain for information we need, including cloud provider credentials for things like hibernation and managed DNS.
+
+ClusterDeployment.Spec.InstallStrategy goes away, and is instead replaced with InstallStrategyRef, which would look like the following:
+
+```yaml
+  installStrategyRef:
+    apiVersion: hive.openshift.io/v1
+    kind: AgentInstallStrategy
+    name: mycluster
+```
+
+InstallStrategyRef is optional, it would not be specified for an adopted cluster.
+
+Hive controllers would then interact with this strategy dynamically with duck typing, similar to Cluster API, Knative, etc.
+
+ClusterDeployment.Spec.Provisioning remains but will eventually be deprecated. Validate this is not used in conjunction with ClusterDeployment.Spec.InstallStrategyRef, must choose one path or the other.
+
+### InstallStrategy Interface
+
+
+Field | Type | Description
+----- | ---- | -----------
+spec.clusterDeploymentRef | LocalObjectReference | Link to the required ClusterDeployment for this install. Bidirectional reference. Common data should live on ClusterDeployment and never be duplicated on InstallStrategy.
+spec.imageSetRef | ClusterImageSetReference | Core mechanism in Hive for choosing what version of OpenShift to install.
+spec.stage | string | Enum string for stage of installation. (Initializing, RequirementsNotMet, Provisioning, RetryProvisioning, Complete, Failed) More details in conditions. *TODO*: Should we just work with conditions? Loss of state in disaster recovery?
+spec.clusterMetadata.adminKubeconfigSecretRef | LocalObjectReference | Admin kubeconfig secret for current install attempt underway, set as soon as known. Copied back to Hive automatically once Ready.
+spec.clusterMetadata.adminPasswordSecretRef | LocalObjectReference | Admin password secret for current install attempt underway, set as soon as known. Copied back to Hive automatically once Ready.
+spec.clusterMetadata.clusterID | string | Cluster UUID generated during install, set as soon as known. Copied back to Hive automatically once Ready.
+spec.clusterMetadata.infraID | string | Cluster infra ID generated during install, set as soon as known. Copied back to Hive automatically once Ready.
+spec.manifestsConfigMapRef | LocalObjectReference | ConfigMap containing manifests to inject into the cluster during installation. *TODO*: Hive technically doesn't need this to be there, do we require it to ensure consistent support or leave it up to install strategies?
```

Comment:
this seems like leak from openshift-install's implementation of the ClusterInstall . So i would be in favor of not requiring this in the contract.

Reply:
Will remove.